components/global/product_switcher_menu/product_switcher_menu.tsx

```diff
@@ -50,6 +57,12 @@ class ProductSwitcherMenu extends React.PureComponent<Props> {
         pluginMenuItems: [],
     };
 
+    async componentDidMount() {
```

Comment:
Nit: Can we remove the `async` keyword? Even though it queues up network requests, we don't need to wait for any of them to complete.